templates/admin/adherent/action_button_switch_user.html.twig

```diff
@@ -1,4 +1,4 @@
-{% if object.enabled %}
+{% if object.enabled and admin.hasAccess('edit') %}
```

Comment:
`is_granted('ROLE_ALLOWED_TO_SWITCH')` will be better

Reply:
no actually, I want to check if he has the role `ROLE_ADMIN_ADHERENTS` or only `ROLE_ADMIN_ADHERENTS_READONLY` (if he has readonly, he has not access on edit for adherents)